blackjax/mcmc/trajectory.py

```diff
@@ -44,6 +44,7 @@
 from blackjax.mcmc.integrators import IntegratorState
 from blackjax.mcmc.proposal import (
     Proposal,
+    hmc_energy,
```

Comment:
this should be defined here, not in `proposal.py`.

The rest of changes here seem good to me.